source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,227 @@
+Transactions
+============
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, multi-document
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-long+} to perform
+**transactions**. :manual:`Transactions </core/transactions/>` allow
+you to run a series of operations that do not change any data until the
+transaction is committed. If any operation in the transaction returns an
+error, the driver cancels the transaction and discards all data changes
+before they ever become visible.
+
+In MongoDB, transactions run within logical **sessions**. A
+:manual:`session </reference/server-sessions/>` is a grouping of related
+read or write operations that you intend to run sequentially. Sessions
+enable :manual:`causal consistency
+</core/read-isolation-consistency-recency/#causal-consistency>` for a
+group of operations or allow you to execute operations in an
+:website:`ACID transaction </basics/acid-transactions>`. MongoDB
+guarantees that the data involved in your transaction operations remains
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can create a new session from a
+``MongoClient`` instance as an ``IClientSession`` type. We recommend that you reuse
+your client for multiple sessions and transactions instead of
+instantiating a new client each time.
+
+.. warning::
+
+   Use an ``IClientSession`` only with the ``MongoClient`` (or associated
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using an
+   ``IClientSession`` with a different ``MongoClient`` results in operation
+   errors.
+
+Methods
+-------
+
+Create an ``IClientSession`` by using either the synchronous ``StartSession()`` or
+the asynchronous ``StartSessionAsync()`` method on your ``MongoClient`` instance. 
+You can then modify the session state by using the method set
+provided by the ``IClientSession`` interface. Select from the following
+:guilabel:`Synchronous Methods` and :guilabel:`Asynchronous Methods`
+tabs to learn about the methods to manage your transaction:
+
+.. tabs::
+
+    .. tab:: Synchronous Methods
+        :tabid: synchronous-methods
+
+        .. list-table::
+            :widths: 40 60
+            :header-rows: 1
+
+            * - Method
+              - Description
+
+            * - ``StartTransaction()``
+              - | Starts a new transaction, configured with the given options, on
+                  this session. Returns an error if there is already
+                  a transaction in progress for the session. To learn more about
+                  this method, see the :manual:`startTransaction() page
+                  </reference/method/Session.startTransaction/>`  in the Server manual.
+                |
+                | **Parameter**: ``TransactionOptions``
+
+            * - ``AbortTransaction()``
```

Comment:
We don't return an error but throw an exception instead. This is applicable for all methods.